pkg/user/RoleGroupService.go

```diff
@@ -525,3 +526,38 @@ func (impl RoleGroupServiceImpl) DeleteRoleGroup(bean *bean.RoleGroup) (bool, er
 
 	return true, nil
 }
+
+func (impl RoleGroupServiceImpl) FetchRolesForGroups(groupNames []string) ([]*bean.RoleFilter, error) {
+	roleGroups, err := impl.roleGroupRepository.GetRoleGroupListByNames(groupNames)
+	if err != nil {
+		impl.logger.Errorw("error while fetching user from db", "error", err)
+		return nil, err
+	}
+	if roleGroups == nil || len(roleGroups) == 0 {
+		return nil, nil
+	}
+
+	var roleGroupIds []int32
+	for _, roleGroup := range roleGroups {
+		roleGroupIds = append(roleGroupIds, roleGroup.Id)
+	}
+
+	roles, err := impl.roleGroupRepository.GetRoleGroupRoleMappingByRoleGroupIds(roleGroupIds)
+	if err != nil {
```

Comment:
shouldnt we hangle pg.ErrNoRows separately? it should return empty array instead of error so that new role can be added to it

Reply:
fixed